provider/pkg/provider/helm_release.go

```diff
@@ -132,6 +134,8 @@ type Release struct {
 	// Manifest map[string]interface{} `json:"manifest,omitempty"`
 	// Names of resources created by the release grouped by "kind/version".
 	ResourceNames map[string][]string `json:"resourceNames,omitempty"`
+	// The checksum of the rendered template (to detect changes)
+	Checksum string `json:"checksum,omitempty"`
```

Comment:
I've not worked on this provider before so I'm curious if this struct can simply be extended? There's no need to mark it as computed or some such? I see no changes to schema in this PR so this is not visible in the schema? 

Does it become visible in diffs? 

I recently had a quick discussion with Fraser about the use of private state. This is something that we use in bridged providers where the resource is allowed to write opaque blobs in to Pulumi state under __meta flag that are generally excluded from treatment by the engine or any of the plumbing, this is purely for the provider to manage. And I was wondering if any native providers do that and how do they do that.

Reply:
It is essentially a computed input (i.e. computed in Check). It does become visible in diffs. I suppose it could be useful as a computed output property. Sorry I don't know what the next step here would be.
```
View in Browser (Ctrl+O): https://app.pulumi.com/eron-pulumi-corp/issue-2118/dev/updates/64

     Type                              Name            Status              Info
     pulumi:pulumi:Stack               issue-2118-dev                      2 warnings
 ~   └─ kubernetes:helm.sh/v3:Release  release         updated (0.10s)     [diff: +checksum]
```

Reply:
That's probably fine then! If we had private state facility you could hide this from users so it's not visible in the diff. But sounds like exposing it is intentional.

Reply:
Agree. I think exposing it would provide more value for our users to understand the diff that is occurring.